src/components/DetailsPage.vue

```diff
@@ -0,0 +1,79 @@
+<script setup lang="ts">
+import { onMounted, ref } from "vue"
+import { useRoute } from "vue-router"
+import { TodoData } from "../todo"
+import { addTodo, deleteTodo, getTodos } from "../apicalls"
+import { useRouter } from "vue-router"
+import { useVuelidate } from "@vuelidate/core"
+import { rules } from "../vuelidateRules"
+
+const currentTodo = ref<TodoData>({
+  isComplete: false,
+    todoName: '',
+    createdAt: '',
+    updatedAt: '',
+    __v: 0,
+    _id: ''
+})
+const route = useRoute()
+const router = useRouter()
+//const todoName= ref<string>('')
+// const applyChanges = () => {
+//   addTodo({todoName: currentTodo.value?.todoName as string, isComplete: currentTodo.value?.isComplete as boolean})
+//   deleteTodo(currentTodo.value?._id as string)
+  
+//   router.push({ path: "/" })
+// }
+
+const vuelidate= useVuelidate(rules, currentTodo)
+
+const applyChanges = () => {
+  vuelidate.value.$touch()
+  console.log(currentTodo.value)
+  if (!vuelidate.value.$invalid) {
+    deleteTodo(currentTodo.value?._id as string)
+    addTodo({
+      todoName: currentTodo.value?.todoName as string,
+      isComplete: currentTodo.value?.isComplete as boolean,
+    })
+    router.push({ path: "/" })
+  }
+}
+
+
+onMounted(async () => {
+  currentTodo.value = (await getTodos(route.query.id as string)) as TodoData
```

Comment:
Why do this casting to TodoData? If this function can possibly return an array which is what it looks like, shouldn't you check for that instead of casting?